Support model runner

glours opened the pull request:
Detect when `models` have been defined in Compose configuration:
* add the env variables to the service containers
* create a new `model-runner` overlay to deploy Model Runner in pod directly inside the Kubernetes cluster
* Use local instance of Docker Model Runner when using the `desktop` overlays to benefit from the host GPU acceleration
* add support also for Helm charts

ndeloof commented on templates/overlays/model-runner/model-runner-deployment.tmpl:
```diff
@@ -0,0 +1,94 @@
+                echo "Pulling model: $model"
+                curl -d "{\"from\": \"$model\"}" "$MODEL_RUNNER"/models/create
+              fi
+            done < /config/models
```
why create a models config map to dynamically iterate on it, while you could just generate this script with models from compose.yaml ?

glours committed:
add Docker Desktop Model Runner support

Support for using the Docker Model Runner instance running on the Docker Desktop host:
- Kubernetes templates: Added model environment variables injection to base deployment template
- Kubernetes overlays: Created desktop overlay that patches deployments to use host.docker.internal:12434
- Helm templates: Added modelRunner.endpoint configuration in values and use it in deployment template

Services with 'models' section in Compose file will have MODEL_URL and MODEL_MODEL environment variables injected automatically.

glours commented:
As discussed last week with @ndeloof, we decided to split the orignal PR in 2:
* This one will only add the support of a local instance of Docker Model Runner within Docker Desktop
* [This 2nd](https://github.com/docker/compose-bridge-transformer/pull/11) manages DMR running as container inside the Kubernetes cluster
